libs/platform/src/lib/components/menu/menu-component/menu.component.ts

```diff
@@ -286,7 +292,7 @@ export class MenuComponent implements OnInit, OnChanges, AfterViewInit, OnDestro
         item.command();
     }
 
-    isMenuGroup(item: MenuItem | MenuGroup): item is MenuGroup {
+    private isMenuGroup(item: MenuItem | MenuGroup): item is MenuGroup {
```

Comment:
I think it would look better if you cast element before putting it as a argument, so usage of this method 
`this.isMenuGroup(<MenuGroup>record)`
Then the method would look like
`private isMenuGroup(item: MenuGroup): boolean`

Reply:
The problem with this is a `record` can be either a `MenuItem` or `MenuGroup` interface, both of which contain fields that are different. So if I were to pass this in the suggested way, `this.isMenuGroup(<MenuGroup>record)` statement would be fine; however, `record` itself can take  either a `MenuItem` or `MenuGroup` and `record.groupItems` within the same code block would fail as `groupItems` belong to `MenuGroup` and not `MenuItem` and we restricted the `record` type to `<MenuGroup>record`.  
Similar problems would arise in the else block, where `record.id` exists on `MenuItem` but not in `MenuGroup`.